File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_PACHECO_BRYAN_ORLANDO/POO/Ejer7/Mascota.py
class Mascota:
    # a) 
    def __init__(self,nombre,tipo,energia):
        self.nombre=nombre
        self.tipo=tipo
        self.energia=energia

    # a) 
    def alimentar(self):
        self.energia+=20
        if self.energia>100:
            self.energia=100
        print(self.nombre,"energia:",self.energia)

    # b) 
    def jugar(self):
        self.energia-=15
        if self.energia<0:
            self.energia=0
        print(self.nombre,"energia:",self.energia)

# c) 
m1=Mascota("Firulais","Perro",50)
m2=Mascota("Michi","Gato",80)


m1.alimentar()  
print("jugando. . . (gasta energia)!!")
m1.jugar()      
m2.alimentar()  
print("jugando. . . (gasta energia)!!")
m2.jugar()      
print(m1.nombre,"y",m2.nombre,"son amigos!!!")

File: repos_estudiantes_TARQUI_primera_entrega/PRADO_FLORES_ARACELY_NAOMI/POO ORIENTADA A OBJETOS/PooSiete/Mascota.java
package PooSiete;
//ejercicio7
public class Mascota {
    
    private String nombre;
    private String tipo;
    private int energia;

    
    public Mascota(String nombre, String tipo, int energia) {
        this.nombre = nombre;
        this.tipo = tipo;
        this.energia = energia;
    }

    
    public void alimentar() {
        energia += 20;
        if (energia > 100) {
            energia = 100;
        }
        System.out.println(nombre + " ha sido alimentado. Energía actual: " + energia);
    }

    
    public void jugar() {
        energia -= 15;
        if (energia < 0) {
            energia = 0;
        }
        System.out.println(nombre + " ha jugado. Energía actual: " + energia);
    }

    
    public void mostrarInfo() {
        System.out.println("Mascota: " + nombre + " | Tipo: " + tipo + " | Energía: " + energia);
    }
    public static void main(String[] args) {
        
        Mascota perro = new Mascota("Firulais", "Perro", 50);
        Mascota gato = new Mascota("Misu", "Gato", 70);

        
        perro.mostrarInfo();
        gato.mostrarInfo();

        
        perro.alimentar();
        gato.alimentar();

        
        perro.jugar();
        gato.jugar();

        
        perro.mostrarInfo();
        gato.mostrarInfo();
    }
}
